app/src/main/java/practicum/getfitla_v3/Nutrition.java

```diff
@@ -1,86 +1,162 @@
+        while (value == "") {
+            try {
+                //check if valid
+                value = row.getJSONArray("c").getJSONObject(cur).optString("v");
+                cur++;
+            } catch (JSONException ex) {
+                //exception for null values
+                cur++;
+            }
+        }
+        //if valid values, add current index and values to the arraylist of answers
+        Values.add(value);
+        Values.add(cur);
+        return Values;
+    }
+    private void processJson(JSONObject object) {
+        try {
+            JSONArray rows = object.getJSONArray("rows");
+            //declare strings outside to be accessible when being added
+            for (int row_id = 0; row_id < rows.length(); ++row_id) {
+                JSONObject row = rows.getJSONObject(row_id);
+                //initialize secondary counter to control for the index
+                int counter = 0;
+                //added try/catch exceptions because of null values in Json files
+                ArrayList Temp;
+                //just null values used to check if a valid value was found
+                String name = "";
+                String shortdesc = "";
+                int image;
+                String prepTime = "";
+                String servingSize = "";
+                String calories = "";
+                String equipment = "";
+                String process = "";
+                String ingredients = "";
+                String rating  = "";
+                String price = "";
+                String x = "";
+                //temporary holding arraylist of answers
+                Temp = add(x, counter,row);
+                //set passed variable to the needed value in the arraylist
+                name = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
+
+                Temp = add(x, counter, row);
+                prepTime = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
+
+                Temp = add(x, counter, row);
+                servingSize = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
+
+                Temp = add(x, counter, row);
+                calories = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
+
+                Temp = add(x, counter, row);
+                equipment = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
+
+                Temp = add(x, counter, row);
+                ingredients = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
+
+                Temp = add(x, counter, row);
+                process = Temp.get(0).toString();
+                counter = Integer.parseInt(Temp.get(1).toString());
 
+                Temp = add(x, counter, row);
+                shortdesc = Temp.get(0).toString();
```

Comment:
I have so many questions about this above section of code. Why is the arraylist named `Temp` (with a capital T, even)? What is `x`? Why do you keep adding `x` to `Temp` then getting the first thing?

Reply:
This was a dumb workaround to return counter (int) and to return the value (string)

Reply:
Hm. Can you do something like (untested):

```java
public String getFromRow(JSONOBject, int counter) {
    try {
        //check if valid
        return row.getJSONArray("c").getJSONObject(counter).optString("v");
    } catch (JSONException ex) {
        //exception for null values
        return "";
    }
}
```

Then in `processJson`:

```java
//initialize secondary counter to control for the index
int counter = 0;

String name = getFromRow(row, counter);
counter++;
String prepTime = getFromRow(row, counter);
counter++;
String servingSize = getFromRow(row, counter);
counter++;
String calories = getFromRow(row, counter);
counter++;
String equipment = getFromRow(row, counter);
counter++;
String ingredients = getFromRow(row, counter);
counter++;
String process = getFromRow(row, counter);
counter++;
String shortdesc = getFromRow(row, counter);
counter++;

// Temporary fix until we have images
int image = R.drawable.nutrition;
```

Reply:
Or if you want to be fancy, the second part can be:

```java
//initialize secondary counter to control for the index
int counter = 0;

String name = getFromRow(row, counter++);
String prepTime = getFromRow(row, counter++);
String servingSize = getFromRow(row, counter++);
String calories = getFromRow(row, counter++);
String equipment = getFromRow(row, counter++);
String ingredients = getFromRow(row, counter++);
String process = getFromRow(row, counter++);
String shortdesc = getFromRow(row, counter++);

// Temporary fix until we have images
int image = R.drawable.nutrition;
```

Reply:
I take everything back. This is the better way:

```java
NutritionItemFormat nutrition = new NutritionItemFormat(
    row_id,
    getFromRow(row, 0), // name
    getFromRow(row, 7), // shortdesc
    image,
    getFromRow(row, 1), // prepTime
    getFromRow(row, 2), // servingSize
    getFromRow(row, 3), // calories
    getFromRow(row, 4), // equipment
    getFromRow(row, 6), // process
    getFromRow(row, 5), // ingredients
    rating,
    price
);
```